### PR TITLE
Split out Developers' guide (for #340)

### PR DESCRIPTION
**Pull request by arporter**
This PR re-structures the PSyclone documentation so that we have `doc/user_guide` and `doc/developer_guide`. Once this is on master psyclone.readthedocs.io will point to the user guide and psyclone-dev.readthedocs.io will point to the developers' guide.

### CODE REVIEW

**Comment by arporter** [comment]
This PR is just for the splitting of the dev and user guides. I'll deal with the reference documentation in a separate Issue.

File: doc/user_guide/pip_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2018, Science and Technology Facilities Council
```
Comment (rupertford) [inline doc/user_guide/pip_requirements.txt]: 2019?

**Comment by rupertford** [comment]
Back to @arporter.

File: doc/README.md [hunk 2]
```diff
@@ -1,14 +1,21 @@
 Generating Documentation
 ========================
 
+PSyclone has both a
+[User Guide](https://psyclone.readthedocs.io/en/stable/) and a
+[Developers' Guide](https://psyclone-dev.readthedocs.io/en/stable/).
+The sources for these are contained in the "user_guide" and
+"developer_guide" directiories, respectively.
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/README.md]: Oops, fixed.

File: doc/README.md [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
+file) before attempting to generate any documentation.
 
 The documentation for PSyclone is created using
 [Sphinx](http://sphinx-doc.org). Documentation can be created in a
 number of formats, such as html, latex and pdf. To see the options
-"cd" to the "doc" directory and type "make" if on a linux system or
-run the "make.bat" script if on a windows system.
+"cd" to either the "user_guide" or "developer_guide" directory and
+type "make" if on a linux system or run the "make.bat" script if on a
+windows system.
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/README.md]: No, I think this is to do with Phil's `extras_require` addition to `setup.py`.

File: doc/developer_guide/conf.py [hunk 4]
```diff
@@ -60,10 +59,10 @@
 # using execfile().
 # 'version' is the short X.Y version and 'release' is the full version,
 # including any alpha/beta/rc tags.
-# We are in the doc directory but need to read version.py from
+# We are in the doc/developer_guide directory but need to read version.py from
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/developer_guide/conf.py]: Oh yes, I completely forgot about that. Done.

File: doc/user_guide/pip_requirements.txt (hunk 1 above)
Comment (arporter) [reply doc/user_guide/pip_requirements.txt]: Done.

**Comment by arporter** [comment]
Well done for spotting the system-specific dev. guide. I've moved that over to the Developers' Guide now. Note that the link to it in the UG currently fails because the version published on Read The Docs does not yet have the new section title in it.

Ready for next review.

File: doc/user_guide/system_specific_setup.rst [hunk 5]
```diff
@@ -1,29 +1,21 @@
...
 
-.. _setup_user:
+This chapter describes the setup for a user of PSyclone. It includes
+all steps necessary to be able to use PSyclone. And while you could
+obviously do some development, none of the required tools for testing
+or documentation creation will be installed.
 
-User set-up
------------
+The :ref:`system_specific_dev_setup` in the Developers' Guide
```
Comment (rupertford) [inline doc/user_guide/system_specific_setup.rst]: Should this be `dev_guide:system_specific_dev_setup`?

**Comment by rupertford** [comment]
Back to @arporter. Just one small question, sorry.

File: doc/user_guide/system_specific_setup.rst (hunk 5 above)
Comment (arporter) [reply doc/user_guide/system_specific_setup.rst]: No, it doesn't have to be. However, it's probably better if it is (clearer that it's an external link) so I've changed it.

**Comment by rupertford** [comment]
Back to @arporter

**Comment by arporter** [comment]
That was weird. Merging on the command line reported no conflicts but Chapter 19 of the User Guide had re-gained its developer-setup section which I've now removed (again).

**Comment by arporter** [comment]
Should be ready to go again now once Travis has reported back.

**Pull request merged**